src/cli/cmds/synth.ts

```diff
@@ -75,4 +116,129 @@ async function fetchValidations(): Promise<ValidationConfig[] | undefined> {
 }
 
 
+async function helmSynthesis(apiVersion: string, chartVersion: string, outdir: string) {
+  const TEMPLATE_WITHOUT_CRDS = 'helm-chart-without-crds';
+  const TEMPLATE_WITH_CRDS = 'helm-chart-with-crds';
+
+  const pkgroot = path.join(__dirname, '..', '..', '..');
+  const templatesDir = path.join(pkgroot, 'templates');
+
+  let templatePath: string;
+
+  const substituteValues = {
+    apiVersion: apiVersion,
+    version: chartVersion,
+    app: path.basename(path.resolve()),
+  };
+
+  // Helm chart structure without CRDs support
+  if (apiVersion === HelmChartApiVersion.V1) {
+
+    if (config.imports && config.imports.length > 0) {
+      throw new Error(`CRDs are not supported in cdk8s for --chart-api-version: ${apiVersion}'. Please use ${HelmChartApiVersion.V2} for using CRDs.`);
+    }
+
+    templatePath = path.join(templatesDir, TEMPLATE_WITHOUT_CRDS);
+  } else {
+    templatePath = path.join(templatesDir, TEMPLATE_WITH_CRDS);
+  }
+
+  try {
+    await sscaff(templatePath, outdir, substituteValues);
+  } catch (error) {
+    throw new Error(`An error occurred during Helm chart creation: ${error}`);
+  }
+
+  await addCrdsToHelmChart(outdir);
+}
+
+async function addCrdsToHelmChart(chartDir: string) {
+  try {
+    const CRD_KIND = 'CustomResourceDefinition';
+    const manifestFiles = await downloadCrds();
+
+    for (const file of manifestFiles) {
+      const parsed = yaml.parseAllDocuments(file);
+
+      let filename;
+      for (const doc of parsed) {
+        const parsedToJson = doc.toJSON();
+        if (
+          parsedToJson.kind === CRD_KIND &&
+          parsedToJson.metadata &&
+          parsedToJson.metadata.name
+        ) {
+          filename = parsedToJson.metadata.name;
+        }
+      }
+
+      if (!filename) {
+        filename = `CRD-${randomUUID}`;
+      }
+
+      console.error('This is where we are - 1');
+
+      fs.writeFileSync(path.join(chartDir, 'crds', filename), file);
+
+      console.error('This is where we are - 2');
+    }
+  } catch (er) {
+    const e = er as any;
+    throw new Error(`error during project initialization: ${e.stack}\nSTDOUT:\n${e.stdout?.toString()}\nSTDERR:\n${e.stderr?.toString()}`);
+  }
+}
+
+async function downloadCrds(): Promise<string[]> {
+  const manifestFiles = [];
+
+  if (config.imports) {
+    for (const crd of config.imports) {
+      let importSpec;
+      try {
+        importSpec = parseImports(crd);
+      } catch (error) {
+        throw new Error(`I caught you: ${error}`);
+      }
```

Comment:
I guess it was for debugging? 

```suggestion
      const importSpec = parseImports(crd);
```

Reply:
Lol, yes, sorry about that. 😶‍🌫️